validate_plan.py

```diff
@@ -0,0 +1,164 @@
+import logging
+import os
+import sys
+from collections.abc import Mapping, Sequence
+from typing import TYPE_CHECKING, Any
+
+from boto3 import Session
+from botocore.config import Config
+from external_resources_io.input import parse_model, read_input_from_file
+from external_resources_io.terraform import (
+    Action,
+    ResourceChange,
+    TerraformJsonPlanParser,
+)
+
+if TYPE_CHECKING:
+    from mypy_boto3_ec2.client import EC2Client
+    from mypy_boto3_ec2.type_defs import SecurityGroupTypeDef, SubnetTypeDef
+else:
+    EC2Client = SubnetTypeDef = SecurityGroupTypeDef = object
+
+from er_aws_msk.app_interface_input import AppInterfaceInput
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger("botocore")
+logger.setLevel(logging.ERROR)
+
+
+class SubnetNotFoundError(Exception):
+    """Subnet not found error"""
+
+    def __init__(self, subnet: str) -> None:
+        super().__init__(f"Subnet {subnet} does not exist")
+
+
+class SecurityGroupNotFoundError(Exception):
+    """Security Group not found error"""
+
+    def __init__(self, sg: str) -> None:
+        super().__init__(f"Security Group {sg} does not exist")
+
+
+class AWSApi:
+    """AWS Api Class"""
+
+    def __init__(self, config_options: Mapping[str, Any]) -> None:
+        self.session = Session()
+        self.config = Config(**config_options)
+
+    @property
+    def ec2_client(self) -> EC2Client:
+        """Gets a boto EC2 client"""
+        return self.session.client("ec2", config=self.config)
+
+    def get_subnet(self, subnet: str) -> SubnetTypeDef:
+        """Get the subnet"""
+        data = self.ec2_client.describe_subnets(
+            SubnetIds=[subnet],
+        )
+        if not data["Subnets"]:
+            raise SubnetNotFoundError(subnet)
+        return data["Subnets"][0]
+
+    def get_security_group(self, subnet: str) -> SecurityGroupTypeDef:
+        """Get the subnet"""
+        data = self.ec2_client.describe_security_groups(
+            GroupIds=[subnet],
+        )
+        if not data["SecurityGroups"]:
+            raise SubnetNotFoundError(subnet)
+        return data["SecurityGroups"][0]
+
+
+class MskPlanValidator:
+    """The plan validator class"""
+
+    def __init__(
+        self, plan: TerraformJsonPlanParser, app_interface_input: AppInterfaceInput
+    ) -> None:
+        self.plan = plan
+        self.input = app_interface_input
+        self.aws_api = AWSApi(config_options={"region_name": "us-east-1"})
```

Comment:
The region should be set from the input data since the vpcs/subnets etc are bound to a region.